### PR TITLE
KAFKA-9068: Updated the javadoc as per implementation

### PR DESCRIPTION
**Pull request by satishbellapu**
Corrected the javadoc as per the session store implementation.

### Committer Checklist (excluded from commit message)
- [*] Verify design and implementation 
- [*] Verify test coverage and CI build status
- [*] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
@satishbellapu Thanks for the PR. It should target `trunk` -- not `2.1` branch. We will cherry-pick to older branches when merging. Thanks.

**Comment by satishbellapu** [comment]
@mjsax do you want me to give a new PR on trunk ?

**Comment by mjsax** [comment]
I tried to change the base branch, but it would pull in other commits. So yes, please open a new PR against `trunk`. Thanks a lot!

File: streams/src/main/java/org/apache/kafka/streams/state/Stores.java [hunk 1]
```diff
@@ -230,10 +230,9 @@ private static WindowBytesStoreSupplier persistentWindowStore(final String name,
     /**
      * Create a persistent {@link SessionBytesStoreSupplier}.
      * @param name              name of the store (cannot be {@code null})
-     * @param retentionPeriod   length ot time to retain data in the store (cannot be negative)
-     *                          Note that the retention period must be at least long enough to contain the
-     *                          windowed data's entire life cycle, from window-start through window-end,
-     *                          and for the entire grace period.
+     * @param retentionPeriod   length to time to retain data in the store (cannot be negative)
```
Comment (ghost) [inline streams/src/main/java/org/apache/kafka/streams/state/Stores.java:233]: This doesn't read well.  Assuming it needs to be as long as `inactivity-gap` plus `grace-period` then the following reads better to me:

```suggestion
     * @param retentionPeriod   length of time to retain data in the store (cannot be negative)
     *                          Note that the retention period must be at least as long as
     *                          inactivity-gap plus grace-period.
```

**Comment by mjsax** [comment]
@satishbellapu Can you please update the PR (ie, open a new one against `trunk`, and also address @WheresAlice comment?) so we can make progress? Closing this PR for now, as it should be against `trunk`. Thanks.

**Pull request closed without merging**